subresource-loading-server.md

```diff
@@ -1,7 +1,47 @@
 # Bundle preloading for servers
 
-<!-- TODO focus this document on resource bundling from the point of view of the server: response behaviour, practical considerations, etc. -->
+This section discusses the proposal from the point of view of primary and intermediate servers.
 
-Considerations about bundle preloading from the point of view of the server. **TODO*
+## Basic operation
+
+As explained in the [overview](./overview.md), the server receives a request for a bundled response that targets the Web bundle file and includes a `Bundle-Preload` header listing the requested resources.
+
+The response from the server will be in the [bundled response](https://github.com/wpack-wg/bundled-responses) format, containing HTTP responses for each of the requested URLs.
+
+The server may return more resources than those originally requested. The browser may choose to cache those resources; it it does so, it should not make those additional resources available to the page until they are themselves included in a `bundlepreload` declaration.
+
+### Response headers
+
+The server response will include the header `Vary: Bundle-Preload`, to signify that the bundle should be requested again for a different value of the `Bundle-Preload` resource list.
+
+`Cache-Control: private` may be used to indicate that the response may be stored only by a browser's cache. This directive is intended as a hint to intermediate servers that they should not try to store many variants of the same bundle.
+
+`Cache-Control: private bundled` (note the new value) may be used to let bundling-aware intermediate servers understand that the data it is not private and may be cached, but probably with a bundling-specific strategy.
+
+<!-- Removed mentions of Bundle-Preload in the response. Add them again if there is a need for it. -->
+
+## Backwards compatibility
+
+By default, the original bundle file stored in the server contains all of the resources that may be requested from it. It is up to the browser and the server to negotiate the eventual transmission of only a subset of those resources.
```

Comment:
Instead of 'by default' - since there's no actual requirement that bundles exist as files, server-side - maybe something like "It is intended that a bundle stored on a static file server contains..."? My wording is not very good there.

Reply:
My idea with "by default" was to link this with the backwards compatibility below: if either the server or the client don't support subsetting, the whole bundle is downloaded.

Reply:
I've rewritten the `Backwards compatibility` section. I hope the new version is a bit clearer.